nomad/structs/structs.go

```diff
@@ -5366,9 +5368,14 @@ func (d *ParameterizedJobConfig) Copy() *ParameterizedJobConfig {
 
 // DispatchedID returns an ID appropriate for a job dispatched against a
 // particular parameterized job
-func DispatchedID(templateID string, t time.Time) string {
+func DispatchedID(prefix, templateID string, t time.Time) string {
 	u := uuid.Generate()[:8]
-	return fmt.Sprintf("%s%s%d-%s", templateID, DispatchLaunchSuffix, t.Unix(), u)
+
+	if prefix == "" {
+		prefix = fmt.Sprint(t.Unix())
+	}
+
+	return fmt.Sprintf("%s%s%s-%s", templateID, DispatchLaunchSuffix, prefix, u)
```

Comment:
I wrote a batch job from this branch and wanted to mess around a bit with some options for how I might want to use interpolation here. Here's an example job with a `template_prefix`:

```hcl
job "example" {
  type        = "batch"
  datacenters = ["dc1"]

  parameterized {
    payload         = "required"
    meta_required   = ["foo"]
    template_prefix = "foo-${meta.foo}"
  }

  group "group" {
    task "task" {

      driver = "raw_exec"

      config {
        command = "/bin/sh"
        args    = ["-c", "cat local/payload.txt; sleep 1"]
      }

      dispatch_payload {
        file = "payload.txt"
      }

    }
  }
}
```

If I dispatch this with `nomad job dispatch -meta foo=bar example ./payload.txt`, I'd want that to be interpolated into something like `example/dispatch-foo-bar-b87be30b`. Right now that's not getting interpolated so of course we get the following instead:

```
$ nomad job status
ID                                               Type                 Priority  Status   Submit Date
example                                          batch/parameterized  50        running  2022-09-21T09:13:46-04:00
example/dispatch-foobar-${meta.foo}-b87be30b  batch                50        dead     2022-09-21T09:13:48-04:00
```

The tricky bit here is that the `${meta.foo}` syntax is HCL, and the HCL is getting rendered into JSON long before it hits the scheduler so we can't use HCL interpolation here. We should be able to support go templating though?